dcmdata/libsrc/dcdatset.cc

```diff
@@ -647,24 +647,47 @@ OFCondition DcmDataset::loadFileUntilTag(const OFFilename &fileName,
     /* check parameters first */
     if (!fileName.isEmpty())
     {
-        /* open file for input */
-        DcmInputFileStream fileStream(fileName);
+        if (*fileName.getCharPointer() == '-')
```

Comment:
I don't understand why you copied & pasted this big block.

Why don't you just do:

```
DcmInputStream *fileStream; // parent from class hierarchy
if stdin
   fileStream = new DcmStdinStream(fileName)
else
   fileStream = new DcmInputFileStream(fileName)
```

This was the reading code chunk would be common.